src/test/java/com/waverleysoftware/SensorTests.java

```diff
@@ -1,20 +1,29 @@
 package com.waverleysoftware;
 
+import com.waverleysoftware.model.Sensor;
+import lombok.extern.slf4j.Slf4j;
 import org.testng.annotations.Test;
 
+import static java.util.Arrays.asList;
+
+/**
+ * A simple model class for sensor testing.
+ */
+@Slf4j
 public class SensorTests extends BaseTest {
-    @Test(priority = 4)
-    public void sensorShouldBeAdded() {
-        System.out.println("Sensor added");
+
+    @Test(dataProvider = "getSensorData", dataProviderClass = DataSuppliers.class)
+    public void sensorShouldBeAdded(final Sensor... sensors) {
+        SOFTASSERT.assertThat(sensors[0].getType().toString().toLowerCase()).as("Sensor type").isEqualTo("embedded");
```

Comment:
it's much easier to compare type with enum value directly to avoid redundant conversions... besides that, you've mixed expected with actual... `sensors` arrays is passed via data supplier, so it's smth you expect to get as a final result...